### PR TITLE
add crio support

### PR DESCRIPTION
**Pull request by vrothberg**
container-feeder defaults to loading the images into the Docker daemon,
which can be changed by setting the "feeder-type" field in the
/etc/container-feeder.json config to "crio".  The loading of the images
to the graphroot of crio is achieved by using the libpod library, adding
more dependencies to the vendor directory.

Notice, that this change implies some heavy refactorings.  Now, we have
a more generalized design and a type hiearchry that will ultimately ease
maintenance.  For packaging purposes a default container-feeder.json
file is added to the root of the project.

Further changes:
  * Add --log-level option to facilitate debugging.
  * Manually update (and fix) containers/image and libpod to load xz
    compressed files. Notice that once libpod has fixed the issue, we
    need to update our ./vendor directory.
  * Update Travis:
    * Use go 1.8.x and 1.9.x
    * Perform fmt checks via govendor
    * Replace go test by a more meaningful compile-test

Signed-off-by: Valentin Rothberg <vrothberg@suse.com>

### CODE REVIEW

File: container-feeder.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+	"feeder-type": "docker"
```
[the remark this answers is not in the document]
Comment (vrothberg) [reply container-feeder.json]: done

File: feeder/crio_feeder.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	defer tmpFile.Close()
+
+	if _, err := io.Copy(tmpFile, xzReader); err != nil {
+		return "", fmt.Errorf("error writing to temporary file: %v", err)
+	}
+
+	return tmpFile.Name(), nil
+}
+
+// LoadImage loads the specified image into docker and returns the image name.
```
[the remark this answers is not in the document]
Comment (vrothberg) [reply feeder/crio_feeder.go]: Done (but changed to `containers/storage`)

File: feeder/feeder.go [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+// NewFeeder returns a new Container Feeder based on the specified type in
+// the container-feeder.json config (default: DockerFeeder)
+func NewFeeder() (Feeder, error) {
+	config, err := loadConfig()
+	if err != nil {
+		log.Errorf("Error loading config file, defaulting to docker feeder: %v", err)
+		config.Target = "docker"
+	}
+
+	if config.Target == "docker" || config.Target == "" {
```
Comment (mjura) [inline feeder/feeder.go]: Maybe we could use `switch` statement here, in future this list can be longer
https://github.com/golang/go/wiki/Switch#switch

Comment (vrothberg) [reply feeder/feeder.go]: Done

**Comment by vrothberg** [comment]
> Why are you removing docker_test.go don't we want to run any unit tests against container-feeder?
@mjura I dec

**Comment by vrothberg** [comment]
At some point, I'd also love to rewrite the `Makefile`. `make build` for instance actually installs it and with the new dependencies on `libpod` we need to add some build tags. `libpod` and the `containers/*` libraries have some scripts to auto-detect btrfs and ostree support etc.

We can do that in a separate PR if needed.

**Pull request merged**